c error: incomplete universal character name \u

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.1.29 0c54ebd
**OS:** Manjaro

**What did you do?**
```v
println('\u')
```

**What did you expect to see?**
Either print nothing or give a compiler error.
 
**What did you see instead?**
```
/tmp/v/testy.15165163179236975449.tmp.c: In function ‘main__main’:
/tmp/v/testy.15165163179236975449.tmp.c:8048:22: error: incomplete universal character name \u
 8048 |  println(tos_lit("\u"));
      |
...
```

--- a/vlib/v/checker/tests/string_escape_u_err_a.out
+++ b/vlib/v/checker/tests/string_escape_u_err_a.out
@@ -1,0 +1,5 @@
+vlib/v/checker/tests/string_escape_u_err_a.vv:2:15: error: `\u` incomplete universal character name
+    1 | fn main() {
+    2 |     println('\u')
+      |               ^
+    3 | }

--- a/vlib/v/checker/tests/string_escape_u_err_a.out
+++ b/vlib/v/checker/tests/string_escape_u_err_a.out
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/string_escape_u_err_a.vv:2:15: error: `\u` incomplete universal character name
+vlib/v/checker/tests/string_escape_u_err_a.vv:2:15: error: `\u` incomplete unicode character value
     1 | fn main() {
     2 |     println('\u')
       |               ^

--- a/vlib/v/checker/tests/string_escape_u_err_a.vv
+++ b/vlib/v/checker/tests/string_escape_u_err_a.vv
@@ -1,0 +1,3 @@
+fn main() {
+    println('\u')
+}

--- a/vlib/v/checker/tests/string_escape_u_err_b.out
+++ b/vlib/v/checker/tests/string_escape_u_err_b.out
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/string_escape_u_err_b.vv:2:15: error: `\u` incomplete universal character name
+vlib/v/checker/tests/string_escape_u_err_b.vv:2:15: error: `\u` incomplete unicode character value
     1 | fn main() {
     2 |     println('\u345')
       |               ^

--- a/vlib/v/checker/tests/string_escape_u_err_b.out
+++ b/vlib/v/checker/tests/string_escape_u_err_b.out
@@ -1,0 +1,5 @@
+vlib/v/checker/tests/string_escape_u_err_b.vv:2:15: error: `\u` incomplete universal character name
+    1 | fn main() {
+    2 |     println('\u345')
+      |               ^
+    3 | }

--- a/vlib/v/checker/tests/string_escape_u_err_b.vv
+++ b/vlib/v/checker/tests/string_escape_u_err_b.vv
@@ -1,0 +1,3 @@
+fn main() {
+    println('\u345')
+}

--- a/vlib/v/scanner/scanner.v
+++ b/vlib/v/scanner/scanner.v
@@ -1050,7 +1050,7 @@ fn (mut s Scanner) ident_string() string {
 				s.text[s.pos + 2] == s.quote || s.text[s.pos + 3] == s.quote || s.text[s.pos + 4] == s.quote ||
 				!s.text[s.pos + 1].is_hex_digit() || !s.text[s.pos + 2].is_hex_digit() || !s.text[s.pos + 3].is_hex_digit() ||
 				!s.text[s.pos + 4].is_hex_digit()) {
-				s.error(r'`\u` incomplete universal character name')
+				s.error(r'`\u` incomplete unicode character value')
 			}
 		}
 		// ${var} (ignore in vfmt mode) (skip \$)

--- a/vlib/v/scanner/scanner.v
+++ b/vlib/v/scanner/scanner.v
@@ -1039,11 +1039,19 @@ fn (mut s Scanner) ident_string() string {
 				s.error(r'cannot use `\x00` (NULL character) in the string literal')
 			}
 		}
-		// Escape `\x`
-		if prevc == slash &&
-			c == `x` && s.count_symbol_before(s.pos - 2, slash) % 2 == 0 && !is_raw && !is_cstr &&
-			(s.text[s.pos + 1] == s.quote || !s.text[s.pos + 1].is_hex_digit()) {
-			s.error(r'`\x` used with no following hex digits')
+		// Escape `\x` `\u`
+		if prevc == slash && !is_raw && !is_cstr && s.count_symbol_before(s.pos - 2, slash) % 2 == 0 {
+			// Escape `\x`
+			if c == `x` && (s.text[s.pos + 1] == s.quote || !s.text[s.pos + 1].is_hex_digit()) {
+				s.error(r'`\x` used with no following hex digits')
+			}
+			// Escape `\u`
+			if c == `u` && (s.text[s.pos + 1] == s.quote ||
+				s.text[s.pos + 2] == s.quote || s.text[s.pos + 3] == s.quote || s.text[s.pos + 4] == s.quote ||
+				!s.text[s.pos + 1].is_hex_digit() || !s.text[s.pos + 2].is_hex_digit() || !s.text[s.pos + 3].is_hex_digit() ||
+				!s.text[s.pos + 4].is_hex_digit()) {
+				s.error(r'`\u` incomplete universal character name')
+			}
 		}
 		// ${var} (ignore in vfmt mode) (skip \$)
 		if prevc == `$` && c == `{` && !is_raw && s.count_symbol_before(s.pos - 2, slash) % 2 == 0 {

